src/main/scala/viper/gobra/frontend/info/implementation/property/Comparability.scala

```diff
@@ -33,16 +34,41 @@ trait Comparability extends BaseProperty { this: TypeInfoImpl =>
   }
 
   lazy val comparableType: Property[Type] = createBinaryProperty("comparable") {
-    case Single(st) => underlyingType(st) match {
-      case t: StructT =>
-        structMemberSet(t).collect {
-          case (_, f: Field) => f.context.symbType(f.decl.typ)
-          case (_, e: Embbed) => e.context.typ(e.decl.typ)
-        }.forall(comparableType)
-
-      case _: SliceT | _: GhostSliceT | _: MapT | _: FunctionT => false
-      case _ => true
+    case Single(st) => st match {
+      case t: TypeParameterT => strictlyComparableType.result(t).holds
+      case _ => underlyingType (st) match {
+        case t: StructT =>
```

Comment:
Why is the TypeParameterT case not inside of the underlying distinction? this looks a bit odd. Maybe you did not defined underlying type correctly.

Reply:
The underlying type of a type parameter would be the underlying type of its type constraint (interface). But the Go spec defines that: "Type parameters are comparable if they are strictly comparable". Further Go's spec defines: "Type parameters are strictly comparable if all types in their type set are strictly comparable."